huaweicloud/services/hss/data_source_huaweicloud_hss_product_infos.go

```diff
@@ -0,0 +1,218 @@
+				Computed:    true,
+				Description: "Whether to enable automatic renewal.",
+			},
+			"version_info": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"periods": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     periodSchema(),
+						},
+					},
+				},
+				Description: "The edition information.",
+			},
+		},
+	}
+}
+
+func dataSourceProductInfosRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		product = "hss"
+		mErr    *multierror.Error
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating HSS client: %s", err)
+	}
+
+	requestPath := client.Endpoint + "v5/{project_id}/product/productdata/offering-infos"
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath += buildProductInfosQueryParams(d, cfg)
+
+	listProductInfosOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", requestPath, &listProductInfosOpt)
+	if err != nil {
+		return diag.Errorf("error retrieving HSS product information: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("data_list", flattenProductInfos(utils.PathSearch("data_list", respBody, make([]interface{}, 0)).([]interface{}))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildProductInfosQueryParams(d *schema.ResourceData, cfg *config.Config) string {
+	epsId := cfg.GetEnterpriseProjectID(d)
+	queryParams := ""
+	if epsId != "" {
+		queryParams = fmt.Sprintf("?enterprise_project_id=%v", epsId)
+	}
+	if v, ok := d.GetOk("site_code"); ok {
+		if queryParams == "" {
+			queryParams = "?"
+		} else {
+			queryParams += "&"
+		}
+		queryParams = fmt.Sprintf("%ssite_code=%v", queryParams, v)
+	}
+	return queryParams
+}
+
+func flattenProductInfos(resp []interface{}) []interface{} {
+	if len(resp) == 0 {
+		return nil
+	}
+
+	rst := make([]interface{}, 0, len(resp))
+	for _, v := range resp {
+		versionInfo := utils.PathSearch("version_info", v, nil)
+		versionList := buildVersionInfoList(versionInfo)
+
+		rst = append(rst, map[string]interface{}{
+			"charging_mode": utils.PathSearch("charging_mode", v, nil),
+			"is_auto_renew": utils.PathSearch("is_auto_renew", v, nil),
+			"version_info":  versionList,
+		})
+	}
+	return rst
+}
+
+func buildVersionInfoList(raw interface{}) []interface{} {
+	if raw == nil {
+		return nil
+	}
+
+	versionMap, ok := raw.(map[string]interface{})
```

Comment:
When parsing data, there is no need to perform type conversion, just use PathSearch.